benchmarks/polys.py

```diff
@@ -195,7 +226,10 @@ def setup(self, n, impl):
             expected = examples.to_expr(expected)
 
         elif impl == 'dense':
-            func = self.get_func_poly(*examples.as_poly())
+            if examples.ring.domain == ZZ_I:
+                func = self.get_func_poly(*examples.as_expr())
+            else:
+                func = self.get_func_poly(*examples.as_poly())
```

Comment:
It should not be necessary to check the domain here like this.

Reply:
If we didn't check the domain then how can we exclude the statement `func = self.get_func_poly(*examples.as_ring)`